### PR TITLE
Feature/mc 10543 async update env

### PR DESCRIPTION
**Pull request by jasondias9**
- Updated the doc for update environment
- removed roles as allowed field to update (this is a v1 concept)

### CODE REVIEW

File: source/includes/administration/_environments.md [hunk 1]
```diff
@@ -215,34 +227,14 @@ curl -X POST "https://cloudmc_endpoint/v2/environments" \
...
-
-The responses' `data` field contains the updated [environment](#administration-environments).
-
 
 <!-------------------- UPDATE ENVIRONMENT -------------------->
 
 
 ### Update environment
 
+Environments are updated asynchronously on the underlying service. When updating an environment any underlying actions performed by the plugin is done asynchronously to the update of the model. The state of the environment will remain the PROVISIONED state. The task returned in the response can be used to track the progress of the asynchronous update.
```
Comment (Shabirmean) [inline source/includes/administration/_environments.md]: `will remain the PROVISIONED state` --> will remain **in** the PROVISIONED state

File: source/includes/administration/_environments.md [hunk 2]
```diff
@@ -167,15 +167,27 @@ curl -X POST "https://cloudmc_endpoint/v2/environments" \
...
+`membership`<br/>*string* | Type of membership of the environment. ALL_ORG_USERS will add every user in the organization to this environment with the default role. MANY_USERS will allow you to  choose the users you want in the environment and assigned them specific roles. *Defaults to MANY_USERS*
+
+Response | &nbsp;
+---------- | -----------
+`taskId`<br/>*UUID* | The id of the task
+`taskStatus`<br/>*string* | The status of the task
+`data`<br/>*[Environment](#administration-environments)* | The information about the created environment
+
 ```
 ```shell
```
Comment (Shabirmean) [inline source/includes/administration/_environments.md:192]: In the rendered view this line is visible as is. Can you check?

File: source/includes/administration/_environments.md [hunk 3]
```diff
@@ -273,7 +259,58 @@ Optional | &nbsp;
 `name`<br/>*string* | The updated name of the environment. Should be unique in the environment and only contain lower case characters, numbers, dashes and underscores.
 `description`<br/>*string* | The updated description of the environment
 `membership`<br/>*string* | Type of membership of the environment. ALL_ORG_USERS will add every user in the organization to this environment with the default role. MANY_USERS will allow you to  choose the users you want in the environment and assigned them specific roles. *Defaults to MANY_USERS*
-`roles`<br/>*Array[[Role](#administration-roles)]* | Update the users roles in the environment. Also, can also update the default role.<br/>*required*: `name`, `users.id`<br/>*optional*: `isDefault`
+
+Response | &nbsp;
```
Comment (Shabirmean) [inline source/includes/administration/_environments.md:263]: Should we show/tell that the response will be a TaskId and we must use it to get the below response?
Or is this what's returned immediate;y?

**Review by Shabirmean — COMMENTED** [review]
some minor comments

File: source/includes/administration/_environments.md (hunk 3 above)
Comment (jasondias9) [reply source/includes/administration/_environments.md:263]: this is what is returned immediately: 
```
{
  data: ...,
  taskId: ...,
  taskStatus: ...,
}
```

Comment (Shabirmean) [reply source/includes/administration/_environments.md:263]: Ohh cool. Didn't see the task info at the end.

**Review by Shabirmean — APPROVED** [review]
LGTM!

File: source/includes/administration/_environments.md [hunk 4]
```diff
@@ -167,16 +167,28 @@ curl -X POST "https://cloudmc_endpoint/v2/environments" \
...
+
+Required | &nbsp;
+-------- | -----------
+`name`<br/>*string* | The name of the new environment. Should be unique in the environment and only contain lower case characters, numbers, dashes and underscores.
+`serviceConnection`<br/>*[ServiceConnection](#administration-service-connections)* | The service connection that the environment should be created in<br/>*required*: `id`
+
+Optional | &nbsp;
+-------- | -----------
+`description`<br/>*string* | The description of the new environment.
+`organization`<br/>*[Organization](#administration-organizations)* | The organization that the environment should be created in. *Defaults to your organization*<br/>*required*: `id`
```
Comment (JamieWulala) [inline source/includes/administration/_environments.md:182]: should this be moved to required? Or if it's default, maybe remove the *required*? Looks confusing to me

Comment (jasondias9) [reply source/includes/administration/_environments.md:182]: the organization is optional - the id is what is required. If you go to the rendered view in github it should be more clear

**Review by JamieWulala — APPROVED** [review]
LGTM

**Pull request merged**